Build: Remove defunct spring-plugins Maven repository from broker build

## Problem

The `StarRocks-Artifacts-Docker-Image` Jenkins build ([#265](https://jenkins.starrocks.com/job/StarRocks-Artifacts-Docker-Image/DC=SGP,buildx=BUILDX_AGENT,distro=ubuntu/265/console)) is failing on the **linux/arm64 broker-builder** stage with the following error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test
  on project hadoop-cos-shaded:
  
  Failed to collect dependencies at
    maven-surefire-plugin:jar:2.12.4 ->
    maven-surefire-common:jar:2.12.4 ->
    org.apache.commons:commons-lang3:jar:3.1:
    
  Could not transfer artifact org.apache.commons:commons-parent:pom:22
    from/to spring-plugins (http://repo.spring.io/plugins-release/):
    Transfer failed: 405 Not Allowed
```

## Root Cause

The `spring-plugins` Maven repository (`http://repo.spring.io/plugins-release/`) has been **permanently closed for public access** by Spring:

| Protocol | Status | Response |
|:---------|:-------|:---------|
| HTTP | **405 Not Allowed** | Blocked |
| HTTPS | **401 Unauthorized** | Requires authentication |

Spring announced this change in [January 2023](https://spring.io/blog/2022/12/14/notice-of-permissions-changes-to-repo-spring-io-january-2023/), stating that all released artifacts should be sourced from **Maven Central** instead.

### Why did this only surface now?

The Docker build uses BuildKit cache mounts (`--mount=type=cache,target=/root/.m2/`). Previously, the cached `.m2` directory on the buildx worker already contained `commons-parent:pom:22`, so Maven never needed to fetch it remotely. A new ECS instance (or cache eviction) caused Maven to re-download from the defunct repository, exposing the issue.

## Impact

- The arm64 broker-builder fails, which **cancels all other parallel build stages** (amd64 fe-builder, amd64 be-builder, amd64 broker-builder, arm64 be-builder)
- The entire `StarRocks-Artifacts-Docker-Image` job fails

## Suggested Fix

Remove the `spring-plugins` repository from the broker's Maven configuration (`fs_brokers/apache_hdfs_broker/` pom.xml or settings.xml). The required dependency `org.apache.commons:commons-parent:pom:22` is available on Maven Central (verified ✅).

--- a/fs_brokers/apache_hdfs_broker/src/pom.xml
+++ b/fs_brokers/apache_hdfs_broker/src/pom.xml
@@ -102,20 +102,7 @@ under the License.
                     <id>cloudera</id>
                     <url>https://repository.cloudera.com/artifactory/cloudera-repos/</url>
                 </repository>
-                <!-- for bdb je -->
-                <repository>
-                    <id>oracleReleases</id>
-                    <url>https://download.oracle.com/maven</url>
-                </repository>
             </repositories>
-
-            <pluginRepositories>
-                <!-- for cup-maven-plugin -->
-                <pluginRepository>
-                    <id>spring-plugins</id>
-                    <url>http://repo.spring.io/plugins-release/</url>
-                </pluginRepository>
-            </pluginRepositories>
         </profile>
     </profiles>
 

